examples/llm/inference/benchmarking/benchmark_deepspeed.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 MosaicML Examples authors
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import time
+
+import deepspeed
+import numpy as np
+import torch
+# You can use this to load the model weights
+from composer.core import get_precision_context
+from omegaconf import OmegaConf as om
+from transformers import AutoTokenizer
+
+from examples.llm.src import COMPOSER_MODEL_REGISTRY
+
+
+def main(config):
+    inf_config = {
+        'replace_with_kernel_inject': True,
+        'dtype': torch.bfloat16,
+        'replace_method': 'auto',
+        'enable_cuda_graph': False,
+        'tensor_parallel': {
+            'tp_size': 0
+        },
+    }
+
+    model = COMPOSER_MODEL_REGISTRY[config.model.name](config.model,
+                                                       config.tokenizer)
+    model.eval()
+
+    tokenizer = AutoTokenizer.from_pretrained(config.tokenizer_name)
+
+    inference_model = deepspeed.init_inference(model, config=inf_config)
+
+    # Checking if deepspeed casts dtypes correctly
+    for n, p in inference_model.named_parameters():
+        print('n is: ', n, p)
```

Comment:
next line is ok, but here you're printing the entire tensor.

Reply:
Can we just check if dtype is as expected here and raise exception if not? No need for prints i think